### ISSUE TITLE
Threadpool / Task Scheduler.

### ISSUE THREAD

**Issue by DeltaGW2** [RaidcoreGG/Nexus#111]
Often times there's stuff like
```cpp
std::thread([this]() {
	this->EventApi->Raise("EV_INPUTBIND_UPDATED");
}).detach();
```

Instead of pushing it to a queue and one thread runs all these events.

**Comment by DeltaRaidcore** [comment]
Worked on as part of https://github.com/RaidcoreGG/Clockwork
